zio-http/src/main/scala/zhttp/service/RequestBodyHandler.scala

```diff
@@ -0,0 +1,23 @@
+package zhttp.service
+import io.netty.channel.ChannelHandler.Sharable
+import io.netty.channel.{ChannelHandlerContext, SimpleChannelInboundHandler}
+import io.netty.handler.codec.http.HttpContent
+import zhttp.http.HttpData.{UnsafeChannel, UnsafeContent}
+
+import scala.annotation.unused
+
+@Sharable
+final class RequestBodyHandler[R](
+  var callback: UnsafeChannel => UnsafeContent => Unit,
+  @unused config: Server.Config[R, Throwable],
+) extends SimpleChannelInboundHandler[HttpContent](false) {
+
+  override def channelRead0(ctx: ChannelHandlerContext, msg: HttpContent): Unit = {
+    callback(UnsafeChannel(ctx))(UnsafeContent(msg))
+  }
+
+  override def handlerAdded(ctx: ChannelHandlerContext): Unit = {
+    ctx.channel().config().setAutoRead(false)
+    ctx.read(): Unit
+  }
+}
```

Comment:
The idea of having `callback` as a function that returns a function is to make sure the first call is made when the handler is added then future calls should just call the returned function with the message.